src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -20,9 +20,10 @@
   "Creates the routes to handle the master's '/v3' routes, which
    includes '/environments' and the non-CA indirected routes. The CA-related
    endpoints are handled separately by the CA service."
-  [request-handler
+  [request-handler-fn
    get-code-id-fn]
-  (let [request-handler-with-code-id (request-core/wrap-with-code-id request-handler get-code-id-fn)]
+  (let [request-handler (comp request-handler-fn request-core/wrap-params-for-jruby)
+        request-handler-with-code-id (request-core/wrap-with-code-id request-handler-fn get-code-id-fn)]
```

Comment:
That's a really good point, and raises an interesting question;

I think you are right, in the context of PE - it would be desirable to only make the function call to retrieve the `code_id` when a JRuby instance is held, thus ensuring that it matches what the catalog will be compiled from.  That's probably a reasonably easy change to make.

However, in OSS Puppet Server, computing `code_is` means shelling-out to an external process., and it seems like it'd be nice to avoid doing that while holding a JRuby instance.

Offhand, I am not sure how feasible it is to satisfy both of those requirements without some pretty major refactoring and possibly duplicating some code into PE.

I lean towards the PE requirement being more important, and so my suggestion would probably be that we:
1. move the call to the function which returns the `current-code-id` so that it happens when a JRuby instance is held.
2. add tests which cover the `code_id` changing due to a File Sync update happening in the background
3. acknowledge that this means we'll be shelling-out to `code-id-command` while holding a JRuby (and perhaps look into how much work it would be to avoid that, although I don't expect that to pan out).
4. not block this PR on those changes since CI is red and this should fix it.

Thoughts, @camlow325 @haus @cprice404 ?